src/Pact/Eval.hs

```diff
@@ -291,12 +395,13 @@ reduce (TObject ps t i) =
 reduce (TBinding ps bod c i) = case c of
   BindLet -> reduceLet ps bod i
   BindSchema _ -> evalError i "Unexpected schema binding"
-reduce t@TModule {} = evalError (_tInfo t) "Module only allowed at top level"
+reduce t@TModule{} = evalError (_tInfo t) "Modules and Interfaces only allowed at top level"
 reduce t@TUse {} = evalError (_tInfo t) "Use only allowed at top level"
 reduce t@TBless {} = evalError (_tInfo t) "Bless only allowed at top level"
 reduce t@TStep {} = evalError (_tInfo t) "Step at invalid location"
 reduce TSchema {..} = TSchema _tSchemaName _tModule _tMeta <$> traverse (traverse reduce) _tFields <*> pure _tInfo
 reduce TTable {..} = TTable _tTableName _tModule _tHash <$> mapM reduce _tTableType <*> pure _tMeta <*> pure _tInfo
+reduce t@TImplements {} = unsafeReduce t -- TODO should not be unsafely reduced
```

Comment:
For some reason I can't continue the conversation from before, so I resolved that and continuing here. The idea with erroring out in `reduce` is that certain forms are handled in special locations, so if they show up here, it means they're in the wrong place:
- `use`: only top-level or module/iface-level (need a better way to describe this usage), handled in `eval` or module load
- `bless`: only module level (error is actually wrong as it stands now, should be "Bless only allowed in module declaration"), handled in module load
- `step`: only within pact, and handled below in `applyPact`
- `module`/`interface`: handled at top level only
- `implements`: same as bless

Reply:
Fixed